packages/components/src/__alpha__/SingleSelect/utils/useFixedOverlayPosition.ts

```diff
@@ -0,0 +1,139 @@
+import { useCallback, useEffect, useLayoutEffect, useState, type RefObject } from 'react'
+
+type LogicalPosition = number | 'auto' | undefined
+
+type Position = {
+  top: LogicalPosition
+  bottom: LogicalPosition
+  insetInlineStart: number
+  maxHeight?: number
+}
+
+type UseFixedOverlayPositionProps = {
+  triggerRef: RefObject<HTMLElement>
+  popoverRef: RefObject<HTMLElement>
+  direction?: 'ltr' | 'rtl'
+  offset?: number
+  preferredPlacement?: 'top' | 'bottom'
+}
+
+/**
+ * Get the nearest scrollable ancestor of an element.
+ */
+function getScrollParent(node: HTMLElement | null): HTMLElement | Window {
+  if (!node) return window
+  const regex = /(auto|scroll|overlay)/
+  const getStyle = (el: HTMLElement): CSSStyleDeclaration => getComputedStyle(el)
+
+  let parent: HTMLElement | null = node
+
+  while (parent && parent !== document.body) {
+    const { overflow, overflowY, overflowX } = getStyle(parent)
+    if (regex.test(overflow + overflowY + overflowX)) {
+      return parent
+    }
+    parent = parent.parentElement
+  }
+
+  return window
+}
+
+/**
+ * Hook to calculate and update the position of an overlay rendered in the top layer.
+ */
+export function useFixedOverlayPosition({
+  triggerRef,
+  popoverRef,
+  direction = 'ltr',
+  offset = 4,
+  preferredPlacement = 'bottom',
+}: UseFixedOverlayPositionProps): Position {
+  const [position, setPosition] = useState<Position>({
+    top: undefined,
+    bottom: undefined,
+    insetInlineStart: 0,
+    maxHeight: undefined,
+  })
+
+  const updatePosition = useCallback(() => {
```

Comment:
Was css anchor positioning part of another card? I have it working on my local but couldn't see it in the epic